waspc/src/Wasp/Generator/Job/Process.hs

```diff
@@ -79,9 +81,12 @@ runProcessAsJob process jobType chan =
 
       return exitCode
 
+    -- NOTE(shayne): We use interruptProcessGroupOf instead of terminateProcess because many
+    -- processes we run will spawn child processes, which themselves may spawn child processes.
+    -- We want to ensure the entire process chain is stopped.
     terminateStreamingProcess streamingProcessHandle = do
       let processHandle = CP.streamingProcessHandleRaw streamingProcessHandle
-      P.terminateProcess processHandle
+      P.interruptProcessGroupOf processHandle
```

Comment:
Nice! Just to check, what if process is not in a group? This will still successfully take care of it?

Reply:
I have to do a bit more verification on those scenarios I think sadly. We did find one issue where if you explicitly set `P.create_group = True` it causes `db migrate-dev` to hang for some reason. So still more testing to do on this sadly, but I think it is still the right direction. I will keep working on it in the background and open another round of Linux testing once I test/address a few items. Thanks! 👍🏻